packages/gatsby-source-contentful/src/__tests__/gatsby-node.js

```diff
@@ -0,0 +1,346 @@
+      store,
+      getNodes,
+      getNode,
+      reporter,
+      createNodeId,
+      cache,
+      getCache,
+    })
+
+    const updatedBlogPost = getNode(updatedBlogPostId)
+    const updatedBlogPostNl = getNode(updatedBlogPostIdNl)
+
+    expect(updatedBlogPost.title).toBe(blogPostEntry.fields.title[`en-US`])
+    expect(updatedBlogPostNl.title).toBe(blogPostEntry.fields.title[`en-US`])
+    expect(updatedBlogPost.tags).toEqual(blogPostEntry.fields.tags[`en-US`])
+    expect(updatedBlogPostNl.tags).toEqual(blogPostEntry.fields.tags[`en-US`])
+    expect(updatedBlogPost[`author___NODE`]).toEqual(
+      normalize.makeId({
+        spaceId: blogPostEntry.sys.space.sys.id,
+        id: blogPostEntry.fields.author[`en-US`].sys.id,
+        defaultLocale: `en-US`,
+        currentLocale: `en-US`,
+      })
+    )
+    expect(updatedBlogPostNl[`author___NODE`]).toEqual(
+      normalize.makeId({
+        spaceId: blogPostEntry.sys.space.sys.id,
+        id: blogPostEntry.fields.author[`en-US`].sys.id,
+        defaultLocale: `en-US`,
+        currentLocale: `nl`,
+      })
+    )
+  })
+
+  it(`should remove a blogpost and update linkedNodes`, async () => {
+    const removedBlogPostId = `a1b2c2345def___c5Q1RtFHGRDbvbY5rZbaaZP`
+    const removedBlogPostIdNl = `a1b2c2345def___c5Q1RtFHGRDbvbY5rZbaaZP___nl`
+
+    fetch
+      .mockReturnValueOnce(startersBlogFixture.initialSync)
+      .mockReturnValueOnce(startersBlogFixture.createBlogPost)
+      .mockReturnValueOnce(startersBlogFixture.removeBlogPost)
+
+    // initial sync
+    await gatsbyNode.sourceNodes({
+      actions,
+      store,
+      getNodes,
+      getNode,
+      reporter,
+      createNodeId,
+      cache,
+      getCache,
+    })
+
+    // create blog post
+    await gatsbyNode.sourceNodes({
+      actions,
+      store,
+      getNodes,
+      getNode,
+      reporter,
+      createNodeId,
+      cache,
+      getCache,
+    })
+
+    expect(currentNodeMap.has(removedBlogPostId)).toBeTruthy()
+    expect(currentNodeMap.has(removedBlogPostIdNl)).toBeTruthy()
+    const authorId = getNode(removedBlogPostId)[`author___NODE`]
+    const authorIdNl = getNode(removedBlogPostIdNl)[`author___NODE`]
+
+    // TODO fix
+    // expect(getNode(authorId)[`blog post___NODE`]).toContain(removedBlogPostId)
+    // expect(getNode(authorIdNl)[`blog post___NODE`]).toContain(
+    //   removedBlogPostIdNl
+    // )
+
+    // remove blog post
+    await gatsbyNode.sourceNodes({
+      actions,
+      store,
+      getNodes,
+      getNode,
+      reporter,
+      createNodeId,
+      cache,
+      getCache,
+    })
+
+    expect(currentNodeMap.has(removedBlogPostId)).toBeFalsy()
+    expect(currentNodeMap.has(removedBlogPostIdNl)).toBeFalsy()
+
+    expect(getNode(authorId)[`blog post___NODE`]).not.toContain(
+      removedBlogPostId
+    )
+    expect(getNode(authorIdNl)[`blog post___NODE`]).not.toContain(
+      removedBlogPostIdNl
+    )
+
+    expect(currentNodeMap).toMatchSnapshot()
```

Comment:
Is this actually valuable assertions? (or just something to debug things out?) Created snapshot is 1000 lines long and with snapshots that long, people won't examine it and just update it whenever I think